src/util/secrets/secrets.c

```diff
@@ -1061,14 +1075,12 @@ errno_t sss_sec_get(TALLOC_CTX *mem_ctx,
 
     if (attr_enctype) {
         enctype = sss_sec_str_to_enctype(attr_enctype);
-        ret = local_decrypt(req->sctx, tmp_ctx, attr_secret, enctype, &secret);
+        ret = local_decrypt(req->sctx, tmp_ctx, attr_secret->data,
+                            attr_secret->length, enctype, &secret, &secret_len);
         if (ret) goto done;
     } else {
-        secret = talloc_strdup(tmp_ctx, attr_secret);
-        if (secret == NULL) {
-            ret = ENOMEM;
-            goto done;
-        }
+        secret = talloc_steal(tmp_ctx, attr_secret->data);
```

Comment:
I know this is widely practiced in SSSD sources, but I still don't like this trick.
This relies on the assumption `struct ldb_val` and `ldb_result` don't have a destructor that does something fancy with its fields.
I doubt there are any docs that give such a promise.

Reply:
If this was true, many things in SSSD would go south, indeed. However, the structure is not opaque so it should be free to use unless ldb maintainers go crazy. Anyway, this particular case was removed per your previous suggestion to remove this branch completely.